server-spi/src/main/java/org/keycloak/models/UserSessionProvider.java

```diff
@@ -138,6 +138,14 @@ UserSessionModel createUserSession(String id, RealmModel realm, UserModel user,
     void removeUserSession(RealmModel realm, UserSessionModel session);
     void removeUserSessions(RealmModel realm, UserModel user);
 
+    /**
+     * Remove expired events in all the realms.
+     * Implementation doesn't need to propagate removal of expired userSessions to userSessionPersister. Cleanup on persister will be called separately
+     */
+    default void removeAllExpired(KeycloakSession session) {
```

Comment:
Is it necessary to introduce `KeycloakSession` parameter? `UserSessionProvider` instance is always aware of the session it was created with. Seems like this would introduce an exception to the rest of the methods.